src/porepy/numerics/ad/grid_operators.py

```diff
@@ -412,6 +438,12 @@ def __repr__(self) -> str:
         )
         return s
 
+    def __str__(self) -> str:
```

Comment:
We are introducing a lot of very similar __str__ methods. Would it be possible to do this by inheritance from Operator, using a method that converts the class type to string?

Reply:
There are subtle, but important, nuances between the different string methods. For instance, the close relation to the class name applies to many, but far from all, string methods. I think it is better to spell it out for each class; at least as long as the whole Ad concept is under heavy development.